File: frontend/src/utils/validaciones.js
// TODO: Implementar validaciones
// - validarCedulaEcuatoriana(cedula): Algoritmo módulo 10
// - validarEmail(email): Validar formato email
// - validarPassword(password): Mínimo 6 caracteres

export const validarCedulaEcuatoriana = (cedula) => {
  const limpia = String(cedula).replace(/[\s-]/g, '');

  if (!/^\d{10}$/.test(limpia)) {
    return { valida: false, mensaje: 'La cédula debe tener 10 dígitos numéricos' };
  }

  const provincia = parseInt(limpia.substring(0, 2), 10);
  if (provincia < 1 || provincia > 24) {
    return { valida: false, mensaje: 'Código de provincia inválido' };
  }

  const tercerDigito = parseInt(limpia[2], 10);
  if (tercerDigito > 6) {
    return { valida: false, mensaje: 'Cédula inválida' };
  }

  const coeficientes = [2, 1, 2, 1, 2, 1, 2, 1, 2];
  const digitoVerificador = parseInt(limpia[9], 10);

  const suma = coeficientes.reduce((acc, coef, i) => {
    let valor = parseInt(limpia[i], 10) * coef;
    if (valor > 9) valor -= 9;
    return acc + valor;
  }, 0);

  const modulo = suma % 10;
  const resultado = modulo === 0 ? 0 : 10 - modulo;

  if (resultado !== digitoVerificador) {
    return { valida: false, mensaje: 'Cédula inválida' };
  }

  return { valida: true, mensaje: 'Cédula válida' };
};

export const validarEmail = (email) => {
  const regex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
  return regex.test(email);
};

export const validarPassword = (password) => {
  if (password.length < 6) {
    return { valida: false, mensaje: 'La contraseña debe tener al menos 6 caracteres' };
  }
  return { valida: true, mensaje: 'Contraseña válida' };
};

File: backend/app/utils/validaciones.py
def validar_cedula_ecuatoriana(cedula):
    """
    Valida una cédula ecuatoriana usando el algoritmo de Módulo 10.
    Retorna un diccionario con 'valida' (bool) y 'mensaje' (str).
    """
    # 1. Debe ser una cadena de texto
    if not isinstance(cedula, str):
        cedula = str(cedula)

    # 2. Debe tener exactamente 10 dígitos y ser numérica
    if not cedula.isdigit() or len(cedula) != 10:
        return {'valida': False, 'mensaje': 'La cédula debe tener 10 dígitos numéricos'}

    # 3. Los dos primeros dígitos representan la provincia (01-24)
    provincia = int(cedula[0:2])
    if provincia < 1 or provincia > 24:
        return {'valida': False, 'mensaje': 'El código de provincia no es válido'}

    # 4. El tercer dígito debe ser menor a 6 para cédulas de persona natural
    tercer_digito = int(cedula[2])
    if tercer_digito >= 6:
        return {'valida': False, 'mensaje': 'El tercer dígito de la cédula no es válido'}

    # 5. Algoritmo de Módulo 10
    coeficientes = [2, 1, 2, 1, 2, 1, 2, 1, 2]
    suma = 0

    for i in range(9):
        digito = int(cedula[i]) * coeficientes[i]
        if digito >= 10:
            digito -= 9
        suma += digito

    digito_verificador_calculado = (10 - (suma % 10)) % 10
    digito_verificador_real = int(cedula[9])

    if digito_verificador_calculado != digito_verificador_real:
        return {'valida': False, 'mensaje': 'El dígito verificador no coincide'}

    return {'valida': True, 'mensaje': 'Cédula válida'}